include/mxnet/lib_api.h

```diff
@@ -1111,16 +1197,49 @@ extern "C" {
 
     // create a vector of tensors for inputs
     std::vector<MXTensor> inputs(num_in);
+    // create a vector for sparse inputs
+    std::vector<MXInSparse> in_sparse(num_in);
+
     for (int i = 0; i < num_in; i++) {
-      inputs[i].setTensor(indata[i], (MXDType)intypes[i], inshapes[i], indims[i],
-                          inIDs[i], {indev_type[i], indev_id[i]});
+      // Dense representation.
+      if(!in_indices_shapes) {
```

Comment:
dont we need to check this on each input? there could be a mix of sparse/dense

Reply:
Fixed.